### ISSUE TITLE
Somewhat confusing documentation on NuGet csbindgen

### ISSUE THREAD

**Issue by fakkoweb** [Cysharp/csbindgen#87]
Hello, I am trying to understand what the NuGet csbindgen is used for as I am having troubles to understand this.

First of all, in Unity it would not be easy to install (nuGet is not supported there) but I am assuming this is just an utility (?). The whole CS bindings generation is done from Rust so I am a bit confused on why this is needed.

I tried to install such NuGet however from the IDE (which I know will be overwritten by Unity, but still it is to try it out) and I get:

```Could not install package 'csbindgen 1.9.2'. You are trying to install this package into a project that targets '.NETFramework,Version=v4.7.1', but the package does not contain any assembly references or content files that are compatible with that framework. For more information, contact the package author.```

However there is no clear dependency specified in csbindgen NuGet, so I am confused on why it does not work.

Can I solve this by simply adding the nuget code to my project somehow?

**Comment by Skgland** [comment]
Based on the [Readme](https://github.com/Cysharp/csbindgen?tab=readme-ov-file#grouping-extension-methods) the nuged package is a source generator to group native methods into extension methods.

**Comment by fakkoweb** [comment]
Ok so if I am using a .net 2.0 csproj library I can add it as a nuget no problem. I think the reason for failing might be due to the `using Microsoft.CodeAnalysis;` not being present when working with Unity.

So can you confirm that the NuGet simply contains the following file being built?
https://github.com/Cysharp/csbindgen/blob/main/GroupedNativeMethodsGenerator/GroupedNativeMethodsGenerator.cs

**Comment by neuecc** [comment]
Using Source Generator in Unity requires a special configuration, details are in the official Unity documentation.
https://docs.unity3d.com/Manual/roslyn-analyzers.html

Also, if you install it via NuGetForUnity, it will do those things automatically.
https://github.com/GlitchEnzo/NuGetForUnity

**Comment by fakkoweb** [comment]
> Also, if you install it via NuGetForUnity, it will do those things automatically. https://github.com/GlitchEnzo/NuGetForUnity

Thank you for your answer.

It is very hard to check on the errors, because there is no direct way to access generated .g.cs files :( even if you wide system search on macOS. Only way I found was to F12 on types in Rider.

I was able to run it on a NetStandard2.0 csproj: it builds with the generated extensions.

The same code however on Unity 2022 (NetStandard2.1 support) I get a:
`The name 'Unsafe' does not exist in the current context` on the generated **Unsafe.AsPointer<T>(T)** in the g.cs

I am wondering why. Maybe I am misunderstanding what it means that "standard" support. It seems this cannot be directly used directly in Unity which is a great shame... unless I am missing something.
